runtime/altair/src/lib.rs

```diff
@@ -1672,19 +1672,20 @@ impl pallet_investments::Config for Runtime {
 	type CollectedRedemptionHook = pallet_foreign_investments::CollectedRedemptionHook<Runtime>;
 	type InvestmentId = InvestmentId;
 	type MaxOutstandingCollects = MaxOutstandingCollects;
-	type PreConditions = IsTrancheInvestor<Permissions, Timestamp>;
+	type PreConditions = IsUnfrozenTrancheInvestor<Permissions, Timestamp>;
 	type RuntimeEvent = RuntimeEvent;
 	type Tokens = Tokens;
 	type WeightInfo = weights::pallet_investments::WeightInfo<Runtime>;
 }
 
 /// Checks whether the given `who` has the role
-/// of a `TrancheInvestor` for the given pool.
-pub struct IsTrancheInvestor<P, T>(PhantomData<(P, T)>);
+/// of a `TrancheInvestor` without having `FrozenInvestor` for the given pool
+/// and tranche.
+pub struct IsUnfrozenTrancheInvestor<P, T>(PhantomData<(P, T)>);
```

Comment:
How difficult would it be to move this to runtime/common? is something I always want to do each time I see this 😆

Reply:
Great point, was simple: [09c4b4c](https://github.com/centrifuge/centrifuge-chain/pull/1947/commits/09c4b4c60e4820ecc3940781a3e6c6ccc1a51c55)

Reply:
Awesome! Thanks a lot for this